model/pdata/metrics.go

```diff
@@ -305,3 +305,37 @@ const (
 func (at AggregationTemporality) String() string {
 	return otlpmetrics.AggregationTemporality(at).String()
 }
+
+// NumberDataPointType specifies the type of NumberDataPoint.
+type NumberDataPointType int32
+
+const (
+	NumberDataPointTypeNull NumberDataPointType = iota
+	NumberDataPointTypeInt
+	NumberDataPointTypeDouble
+)
+
+// Type returns the type of the value for this NumberDataPoint.
+// Calling this function on zero-initialized NumberDataPoint will cause a panic.
+func (ms NumberDataPoint) Type() NumberDataPointType {
+	if ms.orig.Value == nil {
+		return NumberDataPointTypeNull
+	}
+	switch ms.orig.Value.(type) {
+	case *otlpmetrics.NumberDataPoint_AsDouble:
+		return NumberDataPointTypeDouble
+	case *otlpmetrics.NumberDataPoint_AsInt:
+		return NumberDataPointTypeInt
+	}
+	return NumberDataPointTypeNull
+}
+
+// NewNumberDataPointValueInt creates a new NumberDataPointValue with the given int64 value.
+func NewNumberDataPointValueInt(v int64) NumberDataPoint {
+	return NumberDataPoint{orig: &otlpmetrics.NumberDataPoint{Value: &otlpmetrics.NumberDataPoint_AsInt{AsInt: v}}}
+}
+
+// NewNumberDataPointValueDouble creates a new NumberDataPointValue with the given float64 value.
+func NewNumberDataPointValueDouble(v float64) NumberDataPoint {
+	return NumberDataPoint{orig: &otlpmetrics.NumberDataPoint{Value: &otlpmetrics.NumberDataPoint_AsDouble{AsDouble: v}}}
+}
```

Comment:
No need for these 2 constructors yet correct?

Reply:
Correct, I can remove until they're needed.